packages/edit-post/src/components/sidebar/template/actions.js

```diff
@@ -49,14 +50,68 @@ function PostTemplateActions() {
 		},
 		[]
 	);
-	const { __unstableSwitchToTemplateMode } = useDispatch( editPostStore );
+	const {
+		__unstableCreateTemplate,
+		__unstableSwitchToTemplateMode,
+	} = useDispatch( editPostStore );
 
 	if ( ! supportsTemplateMode ) {
 		return null;
 	}
 
 	const defaultTitle = __( 'Custom Template' );
 
+	async function onCreateTemplate( event ) {
```

Comment:
I don't like that this moves the async behavior to the component instead of the action. Also, potentially, `setIsBusy` and any local state setter could be run after the component unmounts causing a react warning.
In terms of UX it's indeed better but we're sacrificing a bit in maintainability.

Maybe it's fine for now.

Reply:
And yeah the `isPerforming` sounds like the right solution here.

Reply:
I can try to take a "stab" on the `isPerforming` selector soon. Because of the lack of a similar selector, this is a pattern I've seen around the core in few places.